inst/include/child_model_simulation.hpp

```diff
@@ -440,59 +449,55 @@ void run_calculate_perinatal_transmission_rate(int t,
-    intermediate.children.retained_on_ART * cpars.PMTCT_transmission_rate(0,4,0) +
-    intermediate.children.retained_started_ART * cpars.PMTCT_transmission_rate(0,5,0) +
-    intermediate.children.PMTCT_coverage(6) * cpars.PMTCT_transmission_rate(0,6,0);
-
-  intermediate.children.receiving_PMTCT = intermediate.children.PMTCT_coverage(0) + intermediate.children.PMTCT_coverage(1) + intermediate.children.PMTCT_coverage(2) + intermediate.children.PMTCT_coverage(3) + intermediate.children.retained_on_ART + intermediate.children.retained_started_ART + intermediate.children.PMTCT_coverage(6);
-  intermediate.children.no_PMTCT = 1 -  intermediate.children.receiving_PMTCT;
-  if(intermediate.children.no_PMTCT < 0){
-    intermediate.children.no_PMTCT = 0;
-  }
-
-  //Transmission among women not on treatment
-  if (intermediate.children.num_wlhiv > 0) {
-    intermediate.children.perinatal_transmission_rate += intermediate.children.no_PMTCT *
-      (intermediate.children.prop_wlhiv_lt200 * cpars.vertical_transmission_rate(4,0) +
-      intermediate.children.prop_wlhiv_200to350 * cpars.vertical_transmission_rate(2,0) +
-      intermediate.children.prop_wlhiv_gte350 * cpars.vertical_transmission_rate(0,0));
-  } else {
-    intermediate.children.perinatal_transmission_rate = intermediate.children.perinatal_transmission_rate;
+  // Calculate transmission rate
+  i_hc.retained_on_ART = i_hc.PMTCT_coverage(4);
+  i_hc.retained_started_ART = i_hc.PMTCT_coverage(5);
+
+  // Transmission among women on treatment
+  i_hc.perinatal_transmission_rate = i_hc.PMTCT_coverage(0) * i_hc.optA_transmission_rate +
+                                     i_hc.PMTCT_coverage(1) * i_hc.optB_transmission_rate +
+                                     i_hc.PMTCT_coverage(2) * p_hc.PMTCT_transmission_rate(0, 2, 0) + // SDNVP
+                                     i_hc.PMTCT_coverage(3) * p_hc.PMTCT_transmission_rate(0, 3, 0) + //dual ARV
+                                     i_hc.retained_on_ART * p_hc.PMTCT_transmission_rate(0, 4, 0) +
+                                     i_hc.retained_started_ART * p_hc.PMTCT_transmission_rate(0, 5, 0) +
+                                     i_hc.PMTCT_coverage(6) * p_hc.PMTCT_transmission_rate(0, 6, 0);
+
+  i_hc.receiving_PMTCT = i_hc.PMTCT_coverage(0) + i_hc.PMTCT_coverage(1) +
+                         i_hc.PMTCT_coverage(2) + i_hc.PMTCT_coverage(3) +
+                         i_hc.retained_on_ART + i_hc.retained_started_ART +
+                         i_hc.PMTCT_coverage(6);
+
+  i_hc.no_PMTCT = 1 - i_hc.receiving_PMTCT;
+  i_hc.no_PMTCT = std::max(i_hc.no_PMTCT, 0.0);
+
+  // Transmission among women not on treatment
+  if (i_hc.num_wlhiv > 0) {
+    auto vertical_transmission = i_hc.prop_wlhiv_lt200 * p_hc.vertical_transmission_rate(4, 0) +
+                                 i_hc.prop_wlhiv_200to350 * p_hc.vertical_transmission_rate(2, 0) +
+                                 i_hc.prop_wlhiv_gte350 * p_hc.vertical_transmission_rate(0, 0);
```

Comment:
new variable introduced

Reply:
Suggest a different name for this for clarity. Something like `untreated_vert_transm_rate`. @mwalte10 also weigh in

Reply:
lets do untreated_vertical_tr?

Reply:
done